### ISSUE TITLE
Logic error: "Code: 253. DB::Exception: Received from localhost:9000. DB::Exception: Replica /mydb/0/mytable/replicas/chi-clickhouse-lakepumper-0-0 already exists. (REPLICA_IS_ALREADY_EXIST)"

### ISSUE THREAD

**Issue by PeerXu** [ClickHouse/ClickHouse#38503]
Hi, everyone.

I try to create a `ReplicatedMergeTree` table `mytable` on `Atomic` database `mydb`, and delete `mytable`.

After that, I try to recreate a `RelplicatedMergeTree` table `mytable`, but got an error:

```
Code: 253. DB::Exception: Received from localhost:9000. DB::Exception: Replica /mydb/0/mytable/replicas/chi-clickhouse-lakepumper-0-0 already exists. (REPLICA_IS_ALREADY_EXIST)
```

Expected:

Recreate the same name table on database.

Steps:

```sql
-- create Atomic database
:) create database if not exists mydb on cluster 'lakepumper' engine = Atomic;
-- create ReplicatedMergeTree table
:) create table mydb.mytable (s String) engine=RelplicatedMergeTree('/mydb/0/mytable', 'chi-clickhouse-lakepumper-0-0') order by s;
-- drop table
:) drop table if exists mydb.mytable on cluster 'lakepumper';
-- recreate ReplicatedMergeTree table but failed
:) create table mydb.mytable (s String) engine=RelplicatedMergeTree('/mydb/0/mytable', 'chi-clickhouse-lakepumper-0-0') order by s;
-- OUTPUT: Received exception from server (version 22.5.1):
-- OUTPUT: Code: 253. DB::Exception: Received from localhost:9000. DB::Exception: Replica /mydb/0/mytable/replicas/chi-clickhouse-lakepumper-0-0 already exists. (REPLICA_IS_ALREADY_EXIST)
```

Environments:
```
Clickhouse Version: 22.5.1.2079
Database Engine: Atomic
Table Engine: ReplicatedMergeTree
```

**Comment by filimonov** [comment]
It is expected. See details in https://kb.altinity.com/engines/altinity-kb-atomic-database-engine/

Tldr version: add `SYNC` keyword to drop table statement.
